github/github_test.go

```diff
@@ -1407,6 +1442,27 @@ func TestDo_rateLimit_abuseRateLimitError_retryAfter(t *testing.T) {
 	if got, want := *abuseRateLimitErr.RetryAfter, 123*time.Second; got != want {
 		t.Errorf("abuseRateLimitErr RetryAfter = %v, want %v", got, want)
 	}
+
+	// expect prevention of a following request
+	_, err = client.Do(ctx, req, nil)
+
+	if err == nil {
```

Comment:
```suggestion
	if _, err := client.Do(ctx, req, nil); err == nil {
```

Reply:
I thought it'd be a bit misleading to put it inside the if statement because we use err later (casting to *AbuseRateLimitError), but I fixed it anyway